pkg/apis/serving/v1alpha1/service_types.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright 2019 Kubeflow Community.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ServiceSpec defines the desired state of Service
+type ServiceSpec struct {
+	// If specified, passes name to generated Revision.
```

Comment:
is this the field for user specified revision name?

Reply:
Correct. This is Knative's proposed impl of this. It's not available yet, so we may need to noop